ThePhasesOfCraftship/2_best_practice_first/strategicDesignPart1/exercises/1_RefactoringTo4Tiers/begin/src/dto/StudentRequestDto.ts

```diff
@@ -0,0 +1,40 @@
+import { isUUID } from "../utils"
+import { ErrorExceptionType } from "../constants"
+
+export type StudentRequestDto = {
+  name?: string
+  id?: string
+}
+
+export default function createStudentRequestDto (data: unknown) {
+  if(!data || typeof data !== 'object') {
+    throw new Error(ErrorExceptionType.ValidationError)
+  }
+  
+  const {name, id} = data as Partial<StudentRequestDto>
```

Comment:
Since all keys on the dto are optional already, using Partial is redundant.
```suggestion
  const {name, id} = data as StudentRequestDto
```

Reply:
Yes, I'm not really a ninja at Typescript so I do mistakes like this haha